File: core/src/main/java/net/minestom/server/utils/math/FloatRange.java
package net.minestom.server.utils.math;

public class FloatRange extends Range<Float> {

  public FloatRange(Float minimum, Float maximum) {
    super(minimum, maximum);
  }

  public FloatRange(Float value) {
    super(value);
  }

  /** {@inheritDoc} */
  @Override
  public boolean isInRange(Float value) {
    return value >= this.getMinimum() && value <= this.getMaximum();

  }
}

File: src/main/java/net/minestom/server/utils/math/FloatRange.kt
package net.minestom.server.utils.math

class FloatRange : Range<Float> {
    constructor(minimum: Float, maximum: Float) : super(minimum, maximum) {}
    constructor(value: Float) : super(value) {}

    /** {@inheritDoc}  */
    override fun isInRange(value: Float): Boolean {
        return value >= minimum && value <= maximum
    }
}